### ISSUE TITLE
Questions should include `allowedCharacters` field

### ISSUE THREAD

**Issue by theodormoroianu** [TeamUnibuc/SmartForms#51]
To perform an efficient OCR, questions should include the `allowedCharacters` field.

I think the client should just have a bunch of textboxes ("lowercase", "uppercase", "numeric" etc) the user can check when creating the form.

**Comment by Rpd-Strike** [comment]
@theodormoroianu 
What are the options for the field: `Allowedcharacters`?

**Comment by theodormoroianu** [comment]
It's a string with all of the possible options. 
Due to the similarities between '0', 'o' and 'O', 'z' and 'Z', '1', 'i' and 'I' etc, the OCR can't really tell lowercase, uppercase and digits appart.

The `allowedCharacters` is simply a string containing all of the allowed characters for the given answer (for instance '0123456789 ' for a phone number - please note the space).

**Comment by Rpd-Strike** [comment]
It would be very tedious for someone to enter the set of all permitted characters for even a single text question. I am thinking of making some presets. 

What do you think of these ones?
1. Everything allowed
2. `013456789 `
3. `a-zA-Z `

Just edit this comment if you have suggestions

**Comment by theodormoroianu** [comment]
Yes, that what I was also thinking. The client could have some presets (lowercase, uppercase, digits etc), which are added as checkmarks on each question, and users can select.

**Comment by theodormoroianu** [comment]
@Rpd-Strike How is this issue going?
